src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -212,8 +220,18 @@ operator()(const NodeID, const EdgeID source_edge_id, Intersection intersection)
                     std::find_if(target_intersection.begin() + 1,
                                  target_intersection.end(),
                                  [this, &link_data](const ConnectedRoad &road) {
-                                     return node_based_graph.GetEdgeData(road.turn.eid).name_id ==
-                                            link_data.name_id;
+
+                                     const auto &road_edge_data =
+                                         node_based_graph.GetEdgeData(road.turn.eid);
+
+                                     const auto same_name = !util::guidance::requiresNameAnnounced(
```

Comment:
from the former checks, this check should only succeed on non-empty nameid values as well